.github/workflows/test-build.yaml

```diff
@@ -0,0 +1,65 @@
+name: Test and Build
+
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+
+jobs:
+  test-build:
+    name: test-build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-go@v2
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v1
+        with:
+          languages: go
+
+      - name: Install kubebuilder
+        run: |
+          set -x
+          os=$(go env GOOS)
+          arch=$(go env GOARCH)
+          curl -Ls https://go.kubebuilder.io/dl/2.3.1/${os}/${arch} | tar -xz -C /tmp/
+          sudo mv /tmp/kubebuilder_2.3.1_${os}_${arch} /usr/local/kubebuilder
+          export PATH=$PATH:/usr/local/kubebuilder/bin
+
+      - name: Get dependencies
+        run: go get -t -v -d ./...
+
+      - name: Test
+        run: make test
+
+      - name: Build
+        run: make manager
+
+      - name: Perform CodeQL Analysis
+        uses: github/codeql-action/analyze@v1
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Build on PR
+        uses: docker/build-push-action@v2
+        with:
+          push: false
+          tags: latest
+        if: github.event_name == 'pull_request'
+
+      - name: Build and publish a latest docker image when merge in main
+        uses: docker/build-push-action@v2
+        with:
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+          registry: docker.pkg.github.com
+          repository: ${{ github.repository }}/argocd-progressive-rollout
+          tags: latest
```

Comment:
Should we move latest only when releasing a new tag or something like that? What's the intended use of latest?

Reply:
So you can test a docker image build from the main branch before cutting a release. Do you think is not a valid use case?

Reply:
well it's subject to discussion. for some project, latest is really the latest build, whereas other is basically the latest stable build. I'd expect to be able to use latest with a degree of stability.
Dan's comment about naming it dev sounds good me to, but again, it's just me. happy to discuss it further but doesn't seem overly important at this point in time.

Reply:
Changed to `main` to reflect is built from the main branch. @danmx is ok with that. If it's ok for you we can keep with main, otherwise I'm happy to change to a different name as long as we have the ability to pull a version built from the tip of main.